sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -28,17 +29,29 @@
     private static final String BEARER_TOKEN_PREFIX = "Bearer ";
     private static final String AUTHORIZATION = "Authorization";
     private final ScopeTokenCache cache;
+    private final HttpClient httpClient;
 
     /**
      * Creates KeyVaultCredentialPolicy.
      *
      * @param credential the token credential to authenticate the request
      */
     public KeyVaultCredentialPolicy(TokenCredential credential) {
+        this(credential, null);
+    }
+
+    /**
+     * Creates KeyVaultCredentialPolicy.
+     *
+     * @param credential the token credential to authenticate the request
+     */
+    public KeyVaultCredentialPolicy(TokenCredential credential, HttpClient httpClient) {
```

Comment:
Add some unit tests to cover the changes in this PR.

Reply:
no more changes needed in KV now with option 1 design.